llvm/lib/Transforms/Coroutines/CoroFrame.cpp

```diff
@@ -52,6 +53,16 @@ extern cl::opt<bool> UseNewDbgInfoFormat;
 
 enum { SmallVectorThreshold = 32 };
 
+static std::string getBasicBlockLabel(const BasicBlock *BB) {
+  if (BB->hasName())
+    return BB->getName().str();
+
+  std::string S;
+  raw_string_ostream OS(S);
+  BB->printAsOperand(OS, false);
+  return OS.str().substr(1);
```

Comment:
Not a fan of this way of getting the name. This seems counterintuitive. As the very least I need a comment here telling me what `printAsOperand` is going to do differently than `BB->getName()`. 

I am also against blanket changes from `BasicBlock::getName()` into `getBasicBlockLabel`. The benefit is unclear to me. Can you show me an example where you won't have a name for the basic block but this `printAsOperand` approach works well?

Reply:
I agree, I will remove it. But I want to point out that BB->getName returns an empty string if the BB has no name, same goes for Instructions. For example, BBs with names like 1234: and Instructions with names like %456 do not have a name. So when we create basic blocks with 'BB->getName() + Twine(".from.") + Pred)->getName()' in most cases that produces a BB called ".from."

Reply:
Under the hood, `printAsOperand` uses `SlotTracker`. (https://github.com/llvm/llvm-project/blob/main/llvm/lib/IR/AsmWriter.cpp#L2599-L2629)

However, it's not entirely clear to me that this slot number will be stable across passes. Looking at the code, it's `printAsOperand` creates the `SlotTracker`. For immediate dumping of the code this may work fine but if you do this again in another pass you may get a different number for the same BB because values can be inserted above.

Reply:
In addressing the reviewers concerns I realized why I added this in the first place. When dumping if we use BB->getName and there is no name, it will not print anything. You are right it is not stable across passes, so it should only be used for dumping. I have made that correction. But when dumping it is important to use otherwise the output will not be understandable. For example, without getBasicBlockLabel the output for coro-suspend-crossing will produce

:
   Consumes: 
      Kills:
:
   Consumes:  
      Kills:
:
   Consumes:   
      Kills:
:
   Consumes:    
      Kills:
.from.:
   Consumes:     .from.
      Kills:
.from.1:
   Consumes:    .from.1
      Kills:
:
   Consumes:     .from. .from.1 
      Kills:
:
   Consumes:     .from. .from.1  
      Kills:


with getBasicBlockLabel we see

2:
   Consumes: 2
      Kills:
16:
   Consumes: 2 16
      Kills:
65:
   Consumes: 2 16 65
      Kills:
100:
   Consumes: 2 16 65 100
      Kills:
.from.:
   Consumes: 2 16 65 100 .from.
      Kills:
.from.1:
   Consumes: 2 16 65 .from.1
      Kills:


You can see there is a lot of missing information if we use getName...